### PR TITLE
Kubernetes proxy integration tests.

### PR DESCRIPTION
**Pull request by klizhentas**
This PR contains Kubernetes proxy
integration tests and associated internal changes.

### CODE REVIEW

**Review by russjones — APPROVED** [review]

**Comment by klizhentas** [comment]
retest this please

File: integration/kube_integration_test.go [hunk 1]
```diff
@@ -0,0 +1,535 @@
...
+	SetTestTimeouts(time.Millisecond * time.Duration(100))
+
+	s.priv, s.pub, err = testauthority.New().GenerateKeyPair("")
+	c.Assert(err, check.IsNil)
+
+	s.ports, err = utils.GetFreeTCPPorts(AllocatePortsNum, utils.PortStartingNumber+AllocatePortsNum+1)
+	if err != nil {
+		c.Fatal(err)
+	}
+	s.me, _ = user.Current()
```
Comment (r0mant) [inline integration/kube_integration_test.go]: Do not ignore the error?

File: integration/kube_integration_test.go [hunk 2]
```diff
@@ -0,0 +1,535 @@
...
+
+func (s *KubeSuite) TearDownSuite(c *check.C) {
+	var err error
+	// restore os.Stdin to its original condition: connected to /dev/null
+	os.Stdin.Close()
+	os.Stdin, err = os.Open("/dev/null")
+	c.Assert(err, check.IsNil)
+}
+
+// TestKubeExec tests kuberentes Exec command set
```
Comment (r0mant) [inline integration/kube_integration_test.go]: kuberentes -> kubernetes

**Comment by klizhentas** [comment]
retest this please

**Review by r0mant — APPROVED** [review]

**Comment by klizhentas** [comment]
retest this please

**Comment by klizhentas** [comment]
retest this please

**Comment by klizhentas** [comment]
retest this please

**Comment by klizhentas** [comment]
retest this please

**Comment by klizhentas** [comment]
retest this please

**Pull request merged**